timemachine/md/moves.py

```diff
@@ -95,6 +95,20 @@ def move(self, x: _State) -> _State:
         return x
 
 
+class BatchedMixtureOfMoves(MixtureOfMoves[_State]):
```

Comment:
This seems like a very simple win, though perhaps there is some nuance I am missing. Seems to make a pretty significant improvement in vacuum, especially with larger number of windows

Reply:
Any suggestions for a better name? It doesn't actually batch the mixture of moves, just generates the sequence in a batch.

Reply:
Maybe this is cleaner as a composition of two independent concepts:
* `IteratedMove` (i.e. `for _ in range(num_iterations): x = move(x)`) 
* `MixtureMove` (i.e. `x = np.random.choice(moves)(x)`) 
?

Reply:
Not clear on what you are suggesting exactly? That is kind of how it was before (though IteratedMove was just a loop in HREX), it used the MixtureOfMoves move, but that results in calling `np.random.choice(48)` `48^3` times which becomes expensive. The goal here is to call np.random.choice only a single time per swap proposal.

Reply:
Ahh, thanks for clarifying! Sorry, I didn't catch the full PR context before commenting. This makes sense as you've written it, and the suggestion is not applicable.

The moves themselves are not batched, but the random selections of which move to apply at each iteration are batched. In the case where the cost of applying each move is comparable to a call to `np.random.choice()`, this yields a speed-up.

(Was thinking of a prior-work-inspired water-sampling context, where the moves themselves might be batched as a form of "speculative execution" -- yielding a speed-up when 1 / acceptance fraction is large relative to the parallel width (i.e. when move(x) == x a lot of the time). This form of "batching" would be independent of whether `move` is a `MixtureOfMoves` or not.)

Reply:
Added a comment to try and clarify the intention of this specialized class https://github.com/proteneer/timemachine/pull/1260/commits/5b4ca483e920634a9c84c90640108f384eb39159